File: app/src/main/java/io/github/freewebmovement/igniter/persistence/Path.java
package io.github.freewebmovement.igniter.persistence;

import android.content.Context;

import java.io.File;

import io.github.freewebmovement.igniter.R;

public class Path {
    public static final int CACHE = 0;
    public static final int FILES = 1;

    public File[] dirs = new File[2];
    public String countryMmdb;
    public String clashConfig;
    public String trojanConfig;
    public String trojanConfigList;
    public String exemptedAppList;
    public String caCert;
    public String systemApps;
    Context context;

    Path(Context context) {
        this.context = context;
        dirs[CACHE] = context.getCacheDir();
        dirs[FILES] = context.getFilesDir();
        countryMmdb = get(FILES, context.getString(R.string.country_mmdb_config));
        clashConfig = get(FILES, context.getString(R.string.clash_config));
        trojanConfig = get(FILES, context.getString(R.string.trojan_config));
        trojanConfigList = get(FILES, context.getString(R.string.trojan_list_config));
        exemptedAppList = get(FILES, context.getString(R.string.exempted_app_list_config));
        caCert = get(FILES, context.getString(R.string.ca_cert_config));
        systemApps = get(FILES, context.getString(R.string.system_apps_config));
    }

    public String get(int type, String filename) {
        switch (type) {
            case CACHE:
            case FILES:
                return new File(dirs[type], filename).getPath();
            default:
                return null;
        }
    }

}

File: app/src/main/java/io/github/freewebmovement/igniter/persistence/Path.kt
package io.github.freewebmovement.igniter.persistence

import android.content.Context
import io.github.freewebmovement.igniter.R
import java.io.File

class Path internal constructor(val context: Context) {
    companion object {
        const val CACHE = 0
        const val FILES = 1
    }

    @JvmField
    val dirs = arrayOfNulls<File>(2)
    @JvmField
    var countryMmdb: String? = null
    @JvmField
    var clashConfig: String? = null
    @JvmField
    var trojanConfig: String? = null
    @JvmField
    var trojanConfigList: String? = null
    @JvmField
    var exemptedAppList: String? = null
    @JvmField
    var caCert: String? = null
    @JvmField
    var systemApps: String? = null

    init {
        dirs[CACHE] = context.cacheDir
        dirs[FILES] = context.filesDir
        countryMmdb = get(FILES, context.getString(R.string.country_mmdb_config))
        clashConfig = get(FILES, context.getString(R.string.clash_config))
        trojanConfig = get(FILES, context.getString(R.string.trojan_config))
        trojanConfigList = get(FILES, context.getString(R.string.trojan_list_config))
        exemptedAppList = get(FILES, context.getString(R.string.exempted_app_list_config))
        caCert = get(FILES, context.getString(R.string.ca_cert_config))
        systemApps = get(FILES, context.getString(R.string.system_apps_config))
    }

    fun get(type: Int, filename: String): String? {
        return when (type) {
            CACHE, FILES -> File(dirs[type]!!, filename).path
            else -> null
        }
    }
}
